edu_rover:
# Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
# Controllers ---------------------------------------

#     left_front ()-[III]-()
#                   [III]
#                   [III]
#                ()-[III]-()

  base_left_front_knuckle_controller:
    type: effort_controllers/JointPositionController #position
    joint: left_front_knuckle_joint
    pid: {p: 5.0, i: 1.0, d: 0.0}

  left_front_knuckle_wheel_controller:
    type: effort_controllers/JointVelocityController #velocity
    joint: left_front_wheel_joint
    pid: {p: 0.01, i: 0.0, d: 0.0}

#                ()-[III]-()
#                   [III]
#                   [III]
#      left_back ()-[III]-()

  base_left_back_knuckle_controller:
    type: effort_controllers/JointPositionController #position
    joint: left_back_knuckle_joint
    pid: {p: 5.0, i: 1.0, d: 0.0}

  left_back_knuckle_wheel_controller:
    type: effort_controllers/JointVelocityController #velocity
    joint: left_back_wheel_joint
    pid: {p: 0.01, i: 0.0, d: 0.0}

#         ()-[III]-() right_front 
#            [III]
#            [III]
#         ()-[III]-()

  base_right_front_knuckle_controller:
    type: effort_controllers/JointPositionController #position
    joint: right_front_knuckle_joint
    pid: {p: 5.0, i: 1.0, d: 0.0}

  right_front_knuckle_wheel_controller:
    type: effort_controllers/JointVelocityController #velocity
    joint: right_front_wheel_joint
    pid: {p: 0.01, i: 0.0, d: 0.0}

#         ()-[III]-()
#            [III]
#            [III]
#         ()-[III]-() right_back 

  base_right_back_knuckle_controller:
    type: effort_controllers/JointPositionController #position
    joint: right_back_knuckle_joint
    pid: {p: 5.0, i: 1.0, d: 0.0}

  right_back_knuckle_wheel_controller:
    type: effort_controllers/JointVelocityController #velocity
    joint: right_back_wheel_joint
    pid: {p: 0.01, i: 0.0, d: 0.0}